riscv/insns/c_sspopchk_x5.h

```diff
@@ -0,0 +1,32 @@
+reg_t prv = STATE.prv;
+reg_t xSSE = 0;
+
+if (prv == PRV_M)
+  xSSE = 0;
+else if (prv == PRV_S || prv == PRV_HS)
+  xSSE = get_field(STATE.menvcfg->read(), MENVCFG_SSE);
+else if (p->extension_enabled('S'))
+  xSSE = get_field(STATE.senvcfg->read(), SENVCFG_SSE);
+else
+  xSSE = false;
+
+// For VS-mode
+if (prv == PRV_S && STATE.v)
+  xSSE &= get_field(STATE.henvcfg->read(), HENVCFG_SSE);
+
+if (!xSSE) {
+  #include "c_mop_N.h"
+} else {
+  reg_t ssp = STATE.ssp->read();
+  const auto new_sp = ssp + xlen / 8;
+  uint64_t ss_addr;
+
+  if (xlen == 32)
+    ss_addr = MMU.ss_load<uint32_t>(ssp);
```

Comment:
The variable ss_addr is not accurate since its not the address of the a shadow stack location. 
How about "shadow_return_address"?